jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/UpdateFunctionCodeDialog.kt

```diff
@@ -0,0 +1,138 @@
+
+    override fun getOKAction(): Action = action
+
+    override fun doCancelAction() {
+        LambdaTelemetry.editFunction(project, result = Result.Cancelled)
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        saveSettings()
+        upsertLambdaCode()
+        close(OK_EXIT_CODE)
+    }
+
+    override fun getHelpId(): String? = HelpIds.UPDATE_FUNCTION_CODE_DIALOG.id
+
+    private fun upsertLambdaCode() {
+        if (!okAction.isEnabled) {
+            return
+        }
+        FileDocumentManager.getInstance().saveAllDocuments()
+
+        val functionDetails = FunctionUploadDetails(
+            name = initialSettings.name,
+            handler = view.handlerPanel.handler.text,
+            iamRole = initialSettings.role,
+            runtime = initialSettings.runtime,
+            description = initialSettings.description,
+            envVars = initialSettings.envVariables ?: emptyMap(),
+            timeout = initialSettings.timeout,
+            memorySize = initialSettings.memorySize,
+            xrayEnabled = initialSettings.xrayEnabled,
+            samOptions = SamOptions(
+                buildInContainer = view.buildSettings.buildInContainerCheckbox.isSelected
+            )
+        )
+
+        val element = findPsiElementsForHandler(project, functionDetails.runtime, functionDetails.handler).first()
+        val psiFile = element.containingFile
+        val module = ModuleUtil.findModuleForFile(psiFile)
+            ?: throw IllegalStateException("Failed to locate module for $psiFile")
+        val lambdaBuilder = initialSettings.runtime.runtimeGroup?.let { LambdaBuilder.getInstanceOrNull(it) }
+            ?: throw IllegalStateException("LambdaBuilder for ${initialSettings.runtime} not found")
+
+        val s3Bucket = view.codeStorage.sourceBucket.selected() as String
+
+        val lambdaCreator = LambdaCreatorFactory.create(project, lambdaBuilder)
+
+        val future = lambdaCreator.updateLambda(module, element, functionDetails, s3Bucket, configurationChanged())
+        future.whenComplete { _, error ->
+            when (error) {
+                null -> {
+                    notifyInfo(
+                        project = project,
+                        title = NOTIFICATION_TITLE,
+                        content = message("lambda.function.code_updated.notification", functionDetails.name)
+                    )
+                    LambdaTelemetry.editFunction(project, update = false, result = Result.Succeeded)
+                }
+                is Exception -> {
+                    error.notifyError(project = project, title = NOTIFICATION_TITLE)
+                    LambdaTelemetry.editFunction(project, update = false, result = Result.Failed)
+                }
+            }
+        }
+    }
+
+    // Using an OkAction to force the validation logic to trigger as well
+    private inner class UpdateFunctionOkAction : OkAction() {
```

Comment:
nit: curious why named inner class vs anonymous object?

Reply:
It was inner class before as well, now that there isnt a switch statement powering it, probably fine to go anonymous.